web/themes/custom/usagov/usagov.theme

```diff
@@ -10,13 +10,167 @@ use Drupal\taxonomy\Entity\Term;
  * Place your custom PHP code in this file.
  */
 
-/**
-* This makes the node varialbe available to the templates/html.html.twig template file
-*/
 function usagov_preprocess_html(&$variables) {
-  $variables['node'] = \Drupal::routeMatch()->getParameter('node');
+  $node = \Drupal::routeMatch()->getParameter('node');
+  $variables['node'] = $node;
+
+  // If the FAQ checkbox is checked, then parse the content to find questions and answers
+  if ($node && $node->hasField('field_faq_page') && $node->get('field_faq_page')[0]->value) {
+    $variables['faq'] = generate_faq($node);
+  }
 }
 
+/**
+ * generate_faq takes a node, analyses the body content,
+ * and returns an array of objects containing a question and answer
+ */
+function generate_faq(&$node) {
+  $faq = [];
+
+  // Set up a DOMDocument with the node's body content
+  $node_body = $node->get('body')[0]->value;
+  $dom = new DOMDocument();
+  libxml_use_internal_errors(true);
```

Comment:
Since I opened this file, phpcs is blaring at me that "true" should be "TRUE" (drupal convention).